public/app/plugins/panel/geomap/module.tsx

```diff
@@ -32,23 +37,68 @@ export const plugin = new PanelPlugin<GeomapPanelOptions>(GeomapPanel)
       defaultValue: defaultView.shared,
     });
 
-    builder.addCustomEditor({
-      category: ['Base Layer'],
-      id: 'basemap',
-      path: 'basemap',
-      name: 'Base Layer',
-      editor: BaseLayerEditor,
-      defaultValue: DEFAULT_BASEMAP_CONFIG,
-    });
+    if (config.geomapDisableCustomBaseLayer) {
+      builder.addCustomEditor({
+        category: ['Base Layer'],
+        id: 'basemap',
+        path: 'basemap',
+        name: 'Base Layer',
+        // eslint-disable-next-line react/display-name
+        editor: () => {
+          return <div>The base layer is configured by the server admin.</div>;
+        },
+        defaultValue: DEFAULT_BASEMAP_CONFIG,
+      });
+    } else {
+      // Append the basemap prefix
+      builder.setDefaultConfig({
+        category: ['Base Layer'],
+        beforeChange: ({ path, value }) => {
+          return { path: `basemap.${path}`, value };
+        },
+        valueGetter: (root: any, path: string) => {
+          return lodashGet(root, `basemap.${path}`);
+        },
+      });
 
-    builder.addCustomEditor({
-      category: ['Data Layer'],
-      id: 'layers',
-      path: 'layers',
-      name: 'Data Layer',
-      editor: DataLayersEditor,
-      defaultValue: [defaultMarkersConfig],
-    });
+      addMapLayerEditors(
+        builder,
+        current?.basemap ?? DEFAULT_BASEMAP_CONFIG,
+        true, // only basemaps
+        (value) => ({
+          path: 'basemap',
+          value,
+        })
+      );
+
+      // Append the basemap prefix
+      builder.setDefaultConfig({
+        category: ['Data Layer'],
+        beforeChange: ({ path, value }) => {
```

Comment:
I don't think the `beforeChange` pattern will work -- I need more explicit control

In this case I want to set `layers[0]` and will eventually need to manage/wrap a whole list.